File: javascript/1313-decompress-run-length-encoded-list.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var decompressRLElist = function(nums) {
  const res = [];
  for (let i = 0; i < nums.length; i += 2) {
    res.splice(res.length, 0, ...new Array(nums[i]).fill(nums[i + 1]));
  }

  return res;
};

console.log(decompressRLElist([1, 1, 2, 3]));

File: python/1313-decompress-run-length-encoded-list.py
class Solution:
    def decompressRLElist(self, nums):
        res = []
        it = iter(nums)
        for x in it:
          t = next(it)
          for _ in range(0, x):
            res.append(t)
        
        return res

c = Solution()
print(c.decompressRLElist([1,1,2,3]))
print(c.decompressRLElist([1, 2, 3, 4]))